### PR TITLE
Edit patent feature

### PR DESCRIPTION
**Pull request by parv112281**
#### JIRA LINK ####

https://patent-visualization.atlassian.net/browse/PSV-53

#### CHANGES ####

- Create Edit button in the patent table for each patent

#### DETAILS ####

- Clicking on the edit button will open a dialog where the user can modify the extracted patent details.

#### MANDATORY GIF ####

PR will **NOT** be aproved if no GIF is included 😄
Include small/social giphy inside the parens **()** https://giphy.com/

### CODE REVIEW

File: src/components/EditDataModal/EditDataModal.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import { Input, Button } from 'antd';
+import Modal from 'react-modal';
```
Comment (blam92) [inline src/components/EditDataModal/EditDataModal.js]: Didn't antd modal work? https://ant.design/components/modal/#header

File: src/components/EditDataModal/EditDataModal.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        <Modal
+            {...props}
+            size="lg"
+            aria-labelledby="contained-modal-title-vcenter"
+            centered
+            style={customStyles}
+        >
+            <div>
+                <h3 id="contained-modal-title-vcenter">
+                    Edit Patent Data: {props.patentDetails.patentNumber}
```
Comment (blam92) [inline src/components/EditDataModal/EditDataModal.js]: nit: Pls consolidate strings in `strings.js` and get them via `StringsManager.get()`

File: src/components/PatentVisualizer/PatentVisualizer.js [hunk 3]
```diff
@@ -275,24 +304,25 @@ const PatentVisualizer = props => {
...
-                    <div>Assignee: {details.assignee}</div>
-                    <div>Other relevant data: </div>
-                    <div>Link to PDF</div>
-                    <div>Text extracted</div>
-                    <div className="visualizer__edit-button-container">
-                        <Button type="primary">Edit Data</Button>
-                    </div>
-                </Sider>
+                {details.show &&
+                    <Sider className="site-layout-background" theme="light" width={200} style={{ padding: '20px' }}>
```
Comment (blam92) [inline src/components/PatentVisualizer/PatentVisualizer.js:330]: Ill file a separate story to clean this detail Sider to a separate component and what is all the data we want to show here.

File: src/components/EditDataModal/EditDataModal.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            right: 'auto',
+            bottom: 'auto',
+            marginRight: '-50%',
+            transform: 'translate(-50%, -50%)',
+            width: '50%',
+        }
+    };
+
+    const submitPatentDetails = () => {
+        props.patentDetails.patentAssignees = assignee;
```
Comment (blam92) [inline src/components/EditDataModal/EditDataModal.js]: props should be immutable and not be reassigned. If you need to edit the original patentDetails from PatentVisualizer you should pass the setPatentDetails function as a callback to get called here.

**Review by blam92 — COMMENTED** [review]
Did not run it locally, one issue with changing props, aside from that looks good. 

Im curious why we did not go with ant design as we already have the dependency for it and it would have the matching styles of the rest of the app.

File: src/components/EditDataModal/EditDataModal.js (hunk 2 above)
Comment (parv112281) [reply src/components/EditDataModal/EditDataModal.js]: fixed this

File: src/components/EditDataModal/EditDataModal.js (hunk 4 above)
Comment (parv112281) [reply src/components/EditDataModal/EditDataModal.js]: fixed this

**Review by blam92 — APPROVED** [review]
LGTM

File: src/components/EditDataModal/EditDataModal.js (hunk 1 above)
Comment (parv112281) [reply src/components/EditDataModal/EditDataModal.js]: Didn't know there was an antd modal, made the changes to use that instead

**Pull request merged**